### PR TITLE
[Improve][Connector-V2][Kudu] Unified exception for kudu source & sink connector

### LINKED ISSUE #3554: [Improve][Connector-V2][Kudu] Unified exception for Kudu source & sink connector
sub-task of #2930

### PR DESCRIPTION
**Pull request by lianghuan-xatu**
## Purpose of this pull request

sub-task of https://github.com/apache/incubator-seatunnel/issues/2930

close https://github.com/apache/incubator-seatunnel/issues/3554

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Comment by lianghuan-xatu** [comment]
@TyrantLucifer PTAL!

**Review by EricJoy2048 — DISMISSED** [review]

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/kuduclient/KuduInputFormat.java [hunk 1]
```diff
@@ -79,8 +81,8 @@ public List<ColumnSchema> getColumnsSchemas() {
             keyColumn = schema.getPrimaryKeyColumns().get(0).getName();
             columns = schema.getColumns();
         } catch (KuduException e) {
-            log.warn("get table Columns Schemas Fail.", e);
-            throw new RuntimeException("get table Columns Schemas Fail..", e);
+            log.error("get table Columns Schemas Failed.", e);
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/kuduclient/KuduInputFormat.java]: Do not use `log.error` before you throw exception.

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/kuduclient/KuduInputFormat.java [hunk 2]
```diff
@@ -116,7 +118,7 @@ public static SeaTunnelRow getSeaTunnelRowData(RowResult rs, SeaTunnelRowType ty
             } else if (BasicType.STRING_TYPE.equals(seaTunnelDataType)) {
                 seatunnelField = rs.getString(i);
             } else {
-                throw new IllegalStateException("Unexpected value: " + seaTunnelDataType);
+                throw new KuduConnectorException(KuduConnectorErrorCode.GET_KUDU_VALUE_FAILED, "Unexpected value: " + seaTunnelDataType);
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/kuduclient/KuduInputFormat.java]: CommonErrorCode.UNSUPPORTED_DATA_TYPE is better, remove this unclear error code.

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/kuduclient/KuduInputFormat.java [hunk 3]
```diff
@@ -136,7 +138,8 @@ public SeaTunnelRowType getSeaTunnelRowType(List<ColumnSchema> columnSchemaList)
             }
         } catch (Exception e) {
             log.warn("get row type info exception.", e);
-            throw new PrepareFailException("kudu", PluginType.SOURCE, ExceptionUtils.getMessage(e));
+            throw new KuduConnectorException(KuduConnectorErrorCode.GET_ROW_TYPE_INFO_FAILD, String.format("PluginName: %s, PluginType: %s, Message: %s",
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/kuduclient/KuduInputFormat.java]: CommonErrorCode.TABLE_SCHEMA_GET_FAILED is better, remove this unclear error code.

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/kuduclient/KuduInputFormat.java [hunk 4]
```diff
@@ -136,7 +138,8 @@ public SeaTunnelRowType getSeaTunnelRowType(List<ColumnSchema> columnSchemaList)
             }
         } catch (Exception e) {
             log.warn("get row type info exception.", e);
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/kuduclient/KuduInputFormat.java]: The same as above.

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/kuduclient/KuduInputFormat.java [hunk 5]
```diff
@@ -180,7 +183,7 @@ public KuduScanner getKuduBuildSplit(int lowerBound, int upperBound) {
                     .addPredicate(upperPred).build();
         } catch (KuduException e) {
             log.warn("get the Kuduscan object for each splice exception", e);
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/kuduclient/KuduInputFormat.java]: The same as above.

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/kuduclient/KuduInputFormat.java [hunk 6]
```diff
@@ -191,7 +194,7 @@ public void closeInputFormat() {
                 kuduClient.close();
             } catch (KuduException e) {
                 log.warn("Kudu Client close failed.", e);
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/kuduclient/KuduInputFormat.java]: The same as above.

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/kuduclient/KuduOutputFormat.java [hunk 7]
```diff
@@ -109,10 +112,10 @@ private void transform(PartialRow row, SeaTunnelRow element, Schema schema) {
                         row.addDecimal(columnIndex, (BigDecimal) element.getField(columnIndex));
                         break;
                     default:
-                        throw new IllegalArgumentException("Unsupported column type: " + col.getType());
+                        throw new KuduConnectorException(KuduConnectorErrorCode.TRANSFORM_TO_KUDU_DATA_TYPE_FAILED, "Unsupported column type: " + col.getType());
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/kuduclient/KuduOutputFormat.java]: CommonErrorCode.UNSUPPORTED_DATA_TYPE is better, remove this unclear error code.

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/kuduclient/KuduOutputFormat.java [hunk 8]
```diff
@@ -129,7 +132,7 @@ private void upsert(SeaTunnelRow element) {
             kuduSession.apply(upsert);
         } catch (KuduException e) {
             log.error("Failed to upsert.", e);
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/kuduclient/KuduOutputFormat.java]: The same as above.

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/kuduclient/KuduOutputFormat.java [hunk 9]
```diff
@@ -142,7 +145,7 @@ private void insert(SeaTunnelRow element) {
             kuduSession.apply(insert);
         } catch (KuduException e) {
             log.error("Failed to insert.", e);
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/kuduclient/KuduOutputFormat.java]: The same as above

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/kuduclient/KuduOutputFormat.java [hunk 10]
```diff
@@ -171,7 +174,7 @@ private void init() {
             kuduTable = kuduClient.openTable(kuduTableName);
         } catch (KuduException e) {
             log.error("Failed to initialize the Kudu client.", e);
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/kuduclient/KuduOutputFormat.java]: The same as above

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSource.java [hunk 11]
```diff
@@ -109,7 +112,7 @@ public void prepare(Config config) {
             columnslist = config.getString(KuduSourceConfig.COLUMNS_LIST.key());
             kuduInputFormat = new KuduInputFormat(kudumaster, tableName, columnslist);
         } else {
-            throw new RuntimeException("Missing Source configuration parameters");
+            throw new KuduConnectorException(SeaTunnelAPIErrorCode.CONFIG_VALIDATION_FAILED, "Missing Source configuration parameters");
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSource.java]: Please refer to other case and change the error tip message.

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSource.java [hunk 12]
```diff
@@ -121,7 +124,7 @@ public void prepare(Config config) {
             SeaTunnelRowType seaTunnelRowType = getSeaTunnelRowType(kuduClient.openTable(tableName).getSchema().getColumns());
             rowTypeInfo = seaTunnelRowType;
         } catch (KuduException e) {
-            throw new RuntimeException("Parameters in the preparation phase fail to be generated", e);
+            throw new KuduConnectorException(KuduConnectorErrorCode.GENERATE_KUDU_PARAMETERS_FAILED, e);
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSource.java:133]: Add error message.

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSource.java [hunk 13]
```diff
@@ -158,7 +161,8 @@ private PartitionParameter initPartitionParameter(KuduClient kuduClient, String
                 }
             }
         } catch (KuduException e) {
-            throw new RuntimeException("Failed to generate upper and lower limits for each partition", e);
+            log.error("Failed to generate upper and lower limits for each partition", ExceptionUtils.getMessage(e));
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSource.java]: The same as above.

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSource.java [hunk 14]
```diff
@@ -175,7 +179,8 @@ public SeaTunnelRowType getSeaTunnelRowType(List<ColumnSchema> columnSchemaList)
 
         } catch (Exception e) {
             log.warn("get row type info exception", e);
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSource.java]: The same as above

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSource.java (hunk 14 above)
Comment (lianghuan-xatu) [reply seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSource.java]: Thanks for your advice. I have fixed it!Thx

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSource.java (hunk 13 above)
Comment (lianghuan-xatu) [reply seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSource.java]: I have fixed it!Thx

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSource.java (hunk 12 above)
Comment (lianghuan-xatu) [reply seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSource.java:133]: I have fixed it! Thx

**Review by TyrantLucifer — APPROVED** [review]

**Comment by lianghuan-xatu** [comment]
@EricJoy2048 PTAL. Thx!

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**